DietPi-Sofware | Homebridge: Bringing HomeKit support where there is none

Zer0x00 opened the pull request:
Should close https://github.com/MichaIng/DietPi/discussions/5142

Documentation can be found here: https://github.com/MichaIng/DietPi-Docs/pull/904

Zer0x00 committed:
Feat: Add Homebridge to DietPi-Software

Zer0x00 committed:
Fix: Remove check for never existing folder

Zer0x00 committed:
Feat: Add homebridge test

MichaIng commented on .github/workflows/dietpi-software.bash:
```diff
@@ -223,6 +223,7 @@ Process_Software()
 			#207) Moonlight (CLI), "moonlight" command
 			#208) Moonlight (GUI), "moonlight-qt" command
 			209) aCOMMANDS[i]='restic version';;
+			211) aCOMMANDS[i]='hb-service status > /dev/null' aSERVICES[i]='homebridge' aTCP[i]='8581';;
```
Not necessary to mute the output. It is actually nice to have it in test results, even if it succeeds:
```suggestion
			211) aCOMMANDS[i]='hb-service status' aSERVICES[i]='homebridge' aTCP[i]='8581';;
```
Indeed this test script and workflow is pretty new and I didn't find time to write some documentation about it yet. Actually worth it since everyone can trigger these tests on the own GitHub repo/fork as well.

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -11707,6 +11714,20 @@ _EOF_
+			G_EXEC eval "curl -sSfL '$url' | gpg --dearmor -o /etc/apt/trusted.gpg.d/dietpi-homebridge.gpg --yes"
+
+			# APT list
+			G_EXEC eval "echo 'deb https://repo.homebridge.io stable main' > /etc/apt/sources.list.d/dietpi-homebridge.list"
```
```suggestion
			G_EXEC eval 'echo '\''deb https://repo.homebridge.io stable main'\'' > /etc/apt/sources.list.d/dietpi-homebridge.list'
```
Tiny, I just like strings without variables consequently single-quoted 🙂.

MichaIng commented on .meta/dietpi-survey_report:
```diff
@@ -673,6 +673,7 @@ shopt -s extglob
 	unset -v 'aSOFTWARE_NAME8_19[141]' # Spotify Connect Web
 	unset -v 'aSOFTWARE_NAME8_19[177]' # Firefox Sync Server
 	aSOFTWARE_NAME8_19[210]='MediaWiki'
+	aSOFTWARE_NAME8_19[211]='Homebridge'
```
This needs to go below to the v8.20 block as `aSOFTWARE_NAME8_20[211]='Homebridge'`.

MichaIng committed:
v8.20

- DietPi-Software | Homebridge: Minor coding, add service to dietpi-services, attempt to fix CI service test

MichaIng committed:
v8.20

- DietPi-Software | Homebridge: Disable for RISC-V, as the repo does not provide packages for it yet, and increase delay before checking TCP port in CI workflow

MichaIng reviewed (approved):
Test issue solved by unsetting capabilities.

For some reason, in the test container on `aarch64`, the service takes very long to start up. However, otherwise it works well, also accessing the web UI and creating the initial admin account is all pretty smooth and simple. Could not find any issues when navigating the web UI etc. Merging it now for testing in beta.
